### ISSUE TITLE
Usage of guava and Apache commons libraries

### ISSUE THREAD

**Issue by jjzazuet** [atomix/atomix#262]
Hi. Hope all is well.

Just saw that the first beta for atomix 2.0 is starting to reference guava and apache commons lang.

In the spirit of keeping bloat out of excellent libraries (such as this one) and in hopes for code minimalism, I'm raising this issue so that the usage of Guava and commons-lang can be carefully considered, keeping in mind that the monolithic package structure they encourage can lead to class path hassles and clashes when importing them as transitive dependencies into other project.

In short, if Guava and commons-lang are meant to be a foundational component of Atomix 2.0, by all means keep them. If however, only a few surface features of these libraries are to be used, I would highly encourage to find minimally designed substitutes.

Thanks and keep the code awesome!

**Comment by kuujo** [comment]
I don't mind removing them, but much of the work that is being moved into Atomix has been done with the aid of various helpers from these libraries. To ensure stability when that move is made, the dependencies are being retained during that process. I certainly welcome contributions to help remove the dependencies once refactoring is complete and before the final release. I don't expect there's anything complex enough to necessitate them.

**Comment by jjzazuet** [comment]
Cool, I'll keep an eye on this one. In the meantime, [Vavr](https://github.com/vavr-io/vavr) might be a good replacement for Guava. I'll see if I can come up with a PR. Thanks!

**Comment by kuujo** [comment]
I'm open to options, but TBH I'm not sure this code uses enough of Guava or Apache Commons to say we shouldn't just write a few utility classes. We just need to get the rest of the code being ported from ONOS into this repo and then we can remove all the dependencies.

**Comment by jjzazuet** [comment]
I would agree with having only utility classes. It's a nice way to keep the dependency tree clean. Thanks!
